commitlint.config.js

```diff
@@ -1 +1,54 @@
-module.exports = { extends: ["@commitlint/config-conventional"] };
+// https://github.com/conventional-changelog/commitlint/blob/master/@commitlint/config-conventional/index.js
+// https://github.com/conventional-changelog/commitlint/blob/master/docs/reference-rules.md
+module.exports = {
+  extends: ["@commitlint/config-conventional"],
+  plugins: ["commitlint-plugin-function-rules"],
+  rules: {
+    "type-enum": [
+      2,
+      "always",
+      // in addtition to angular defaults, we add additional type for commits, breaking, breaking_change will force a major release with semantic-release
+      // any body of commit containing BREAKING_CHANGE, BREAKING, etc.. will do too, please refer to the .releaserc.json file
+      [
+        "revert",
+        "feat",
+        "test",
+        "perf",
+        "fix",
+        "chore",
+        "ci",
+        "docs",
+        "breaking_change",
+        "breaking",
+        "refactor",
+      ],
+    ],
+    "subject-case": [1, "always", "lower-case"],
+    // disable it to allow release notes from semantic-release to happen when releasing
+    "footer-max-line-length": [0],
+    "body-max-line-length": [0],
+    // level: disabled to use our own rule with commitlint-plugin-function-rules plugin
+    "scope-case": [0],
+    "function-rules/scope-empty": [
+      2,
+      "always",
+      (parsed) => {
+        const ticketNumberFormat = /^[a-z0-9]+-\d+$/;
```

Comment:
Our branches are in uppercase and it causing issues for example VTA-12345

Reply:
This probably needs updating in the template too

Reply:
I'll change to uppercase. Our branches/tickets are uppercase too. Not sure who initial wanted it lower.
I'll be adding it to the template too.

Reply:
Looking at the historical commit messages on VTA, where you specify the commit message - the commit messages are in lowercase i.e. feat(vta-1312): message ,(even though all the feature branches appear to be uppercase - feature/VTA-1312) - Is it worth making it consistent? Will this have any impact on the DevOps side for example semantic release?

Reply:
So shall I put it back to lowercase then? I have no preference.

Reply:
Do we even want this regex? What value does it add?

Reply:
This is what I was asking earlier. Who has final say on what we all should be doing across all the DVSA repos.